server/shadow/shadow_client.c

```diff
@@ -981,9 +978,7 @@ void* shadow_client_thread(rdpShadowClient* client)
 				shadow_client_send_surface_update(client);
 			}
 
-			EnterSynchronizationBarrier(&(subsystem->barrier), 0);
-
-			while (WaitForSingleObject(UpdateEvent, 0) == WAIT_OBJECT_0);
+			shadow_multiclient_consume(UpdateSubscriber);
```

Comment:
The return value of shadow_multiclient_consume is whether or not the subscriber really consumes the event. It's not cared currently. An explanation would be added, instea :)